common/src/main/java/io/netty/util/internal/NativeLibraryLoader.java

```diff
@@ -371,4 +390,62 @@ private static void closeQuietly(Closeable c) {
     private NativeLibraryLoader() {
         // Utility
     }
+
+    private static final class NoexecVolumeDetector {
+
+        private static boolean canExecuteExecutable(File file)
+                throws ClassNotFoundException, NoSuchMethodException, InvocationTargetException,
+                       IllegalAccessException {
+            if (PlatformDependent.javaVersion() < 7) {
+                // Assume true because we
```

Comment:
Sorry -- was rushing yesterday and didn't re-read my comments.  They should be much clearer now.